source/boundary_temperature/interface.cc

```diff
@@ -167,12 +167,18 @@ namespace aspect
               = std::set<types::boundary_id> (x_fixed_temperature_boundary_indicators.begin(),
                                               x_fixed_temperature_boundary_indicators.end());
 
-            // If model names have been set, but no boundaries on which to use them,
-            // ignore the set values, do not create objects that are never used.
+            // If no fixed temperature boundary indicators have been set, there should be no model_names chosen either.
+            // If that is indeed the case, clear the model_operators vector. Otherwise, raise an exception.
             if (fixed_temperature_boundary_indicators.size() == 0)
               {
-                model_names.clear();
-                model_operators.clear();
+                if (model_names.size() == 0)
+                  model_operators.clear();
+                else
+                  AssertThrow(false,
+                              ExcMessage ("You have indicated that you wish to apply a boundary temperature "
+                                          "model, but the <Fixed temperature boundary indicators> parameter "
+                                          "is empty. Please use this parameter to specify the boundaries "
+                                          "on which the model(s) should be applied."));
```

Comment:
If you write it like this the condition appears in the error message, which is helpful to understand the problem.

```suggestion
                  AssertThrow(model_names.size() == 0,
                              ExcMessage ("You have indicated that you wish to apply a boundary temperature "
                                          "model, but the <Fixed temperature boundary indicators> parameter "
                                          "is empty. Please use this parameter to specify the boundaries "
                                          "on which the model(s) should be applied."));
                                          
                 model_operators.clear();
```